### ISSUE TITLE
broken in webpack 3

### ISSUE THREAD

**Issue by bradennapier** [webpack-contrib/uglifyjs-webpack-plugin#47]
Argument spreading is broken in uglify-js. This worked fine in rc of 3.0.0

```
"uglifyjs-webpack-plugin": "^0.4.6",
```

```
"webpack": "^3.0.0",
```

It is not using concat to spread the arguments when called in a function causing a ton of problems.

```js
Object.keys(selected).reduce((...args) => {
    console.log('reduction: ', args, args.length);
    return handleProcessReduction(selected, ...args);
  }, {});
```

In the above code, args is sent to handleProcessReduction as the second arg only.

Looking at the transpiled code:

```js
return Object.keys(e).reduce(function() {
            for (var r = arguments.length, t = Array(r), n = 0; n < r; n++)
              t[n] = arguments[n];
            return console.log(
              'reduction: ',
              t,
              t.length,
            ), u.apply(void 0, [e, t]); // <---- t should have been spread [e].concat(t)
          }, {});
```

**Comment by evilebottnawi** [comment]
@bradennapier do your use `babel-loader` (in theory he should handle spread)?

**Comment by bradennapier** [comment]
yeah and babel-loader does it fine.  It's not until uglify is applied that it breaks it.

**Comment by evilebottnawi** [comment]
@bradennapier seems `uglify-js` doesn't works as expected, can your test `babel` transpiled on using `uglify-js` standalone, this will allow us to understand whether there is a problem

**Comment by michael-ciniawsky** [comment]
The current version of `uglify-js` in use, doesn't support any ES2015+ (ES6) minification yet, will be fixed by #63  soon
